### PR TITLE
Feat/message class

### PR DESCRIPTION
**Pull request by RLKRo**
# Description

This PR updates dff modules to use `Message` (former `Response`) as type for script responses and requests.

# Checklist

- [x] Updated most of the tests and examples
- [x] Updated examples core3-6, utils1-2
- [ ] I have added comments to my code to help others understand it
- [ ] I have updated the documentation to reflect the changes
- [x] I have performed a self-review of the changes

### CODE REVIEW

File: dff/script/conditions/std_conditions.py [hunk 1]
```diff
@@ -27,7 +27,9 @@ def exact_match(match: Any, *args, **kwargs) -> Callable[[Context, Actor, Any, A
 
     def exact_match_condition_handler(ctx: Context, actor: Actor, *args, **kwargs) -> bool:
         request = ctx.last_request
-        return match == request
+        if request is None:
+            return False
+        return match == request.text
```
Comment (kudep) [inline dff/script/conditions/std_conditions.py:32]: I think text match is a good option for  `regexp`. We can remove `exact_match` at all. What do you think?

File: dff/script/core/actor.py [hunk 2]
```diff
@@ -395,10 +396,10 @@ def validate_script(self, verbose: bool = True):
             response_func = normalize_response(node.response)
             try:
                 response_result = response_func(ctx, actor)
-                if isinstance(response_result, Callable):
+                if not isinstance(response_result, Message):
```
Comment (kudep) [inline dff/script/core/actor.py:399]: Response can be callable in cases when callable response returns `Message`

File: dff/script/core/context.py [hunk 3]
```diff
@@ -153,7 +154,7 @@ def add_request(self, request: Any):
         self.requests[last_index + 1] = request
 
     @validate_arguments
-    def add_response(self, response: Any):
+    def add_response(self, response: Message):
```
Comment (kudep) [inline dff/script/core/context.py:157]: Response can be callable too

**Review by kudep — CHANGES_REQUESTED** [review]

File: dff/script/core/actor.py (hunk 2 above)
Comment (RLKRo) [reply dff/script/core/actor.py:399]: But here we check the type of `response_result` which is the return value of `response_func` (`Callable[[Context, Actor], Message]`), so `response_result` should be `Message`.

File: dff/script/core/context.py (hunk 3 above)
Comment (RLKRo) [reply dff/script/core/context.py:157]: The doc for `Context.responses` says that it stores a history of all responses. So I'm assuming that means if response was callable its result is what should be stored. This is corroborated by the only usage of the function here
```
Actor.__call__
        # create response
        ctx.framework_states["actor"]["response"] = ctx.framework_states["actor"][
            "pre_response_processed_node"
        ].run_response(ctx, self, *args, **kwargs)
        self._run_handlers(ctx, ActorStage.CREATE_RESPONSE, *args, **kwargs)
        ctx.add_response(ctx.framework_states["actor"]["response"])
```
where a value that is passed to the function is a result of `run_response` which returns `Message`;
As well as by the usages of `Context.last_response` (such as `utils/testing/common:check_happy_path`) which assume it has a `Message` type:
```
        candidate_response = ctx.last_response
        if printout_enable:
            print(f"(user) >>> {request}")
            print(f" (bot) <<< {candidate_response}")
```

**Comment by kudep** [comment]
@ruthenian8 can you describe what is it for  
https://github.com/deeppavlov/dialog_flow_framework/blob/d584c6824fe40cea77c8e83c18fe776d58c0c1e6/dff/script/responses/generics.py#L38

I do not get the idea of simpplification

File: dff/script/conditions/std_conditions.py (hunk 1 above)
Comment (kudep) [reply dff/script/conditions/std_conditions.py:32]: I think just remove `exact_match`

File: dff/script/core/actor.py (hunk 2 above)
Comment (kudep) [reply dff/script/core/actor.py:399]: Yes, you are right

File: dff/script/core/context.py (hunk 3 above)
Comment (kudep) [reply dff/script/core/context.py:157]: Yes, you are right. I was in a hurry.

File: examples/script/core/3_responses.py [hunk 4]
```diff
@@ -49,37 +49,39 @@ def cannot_talk_about_topic_response(ctx: Context, actor: Actor, *args, **kwargs
...
     def cannot_talk_about_topic_response(ctx: Context, actor: Actor, *args, **kwargs) -> Any:
-        return response.upper()
+        if response.text is not None:
+            response.text = response.text.upper()
+        return response
 
     return cannot_talk_about_topic_response
 
 
 def fallback_trace_response(ctx: Context, actor: Actor, *args, **kwargs) -> Any:
```
Comment (kudep) [inline examples/script/core/3_responses.py:67]: For response function should return Message type in a signature

**Review by kudep — CHANGES_REQUESTED** [review]

File: dff/script/conditions/std_conditions.py (hunk 1 above)
Comment (RLKRo) [reply dff/script/conditions/std_conditions.py:32]: I don't think this is a good idea. While `exact_match` may not be useful in practice, it is still a very good function to showcase the framework (which is why most of the examples use it).

**Review by kudep — CHANGES_REQUESTED** [review]
- [ ] exact matching
- [ ] comments & docs

File: makefile [hunk 5]
```diff
@@ -29,14 +29,14 @@ venv:
...
 
 format: venv
-	black --line-length=120 --exclude='venv|build|examples' .
+	black --line-length=120 dff tests
 	black --line-length=100 examples
 .PHONY: format
 
 lint: venv
-	flake8 --max-line-length=120 --exclude venv,build,examples .
+	flake8 --max-line-length=120 dff tests
```
Comment (kudep) [inline makefile:37]: ????

File: dff/utils/testing/response_comparers.py [hunk 6]
```diff
@@ -18,9 +15,9 @@ def default_comparer(candidate: Any, reference: Any, _: Context) -> Optional[Any
     return None if candidate == reference else candidate
 
 
-def generics_comparer(candidate: Response, reference: str, _: Context) -> Optional[str]:
+def generics_comparer(candidate: Message, reference: Message, _: Context) -> Optional[str]:
```
Comment (kudep) [inline dff/utils/testing/response_comparers.py:18]: it can be removed

File: examples/script/responses/3_media.py [hunk 7]
```diff
@@ -36,36 +35,36 @@
...
+            RESPONSE: Message(text="Please, send me a picture url"),
             TRANSITIONS: {
                 ("pics", "send_one", 1.1): cnd.regexp(r"^http.+\.png$"),
                 ("pics", "send_many", 1.0): cnd.regexp(r"^http.+\.jpg$"),
                 ("pics", "repeat", 0.9): cnd.true(),
             },
         },
         "send_one": {
-            RESPONSE: Response(text="here's my picture!", image=Image(source=kitten_url)),
+            RESPONSE: Message(text="here's my picture!", image=Image(source=kitten_url)),
```
Comment (kudep) [inline examples/script/responses/3_media.py:56]: For this one we should use attachments too, shouldn’t we?

**Review by kudep — CHANGES_REQUESTED** [review]

**Review by pseusys — APPROVED** [review]
Generally OK, but you could've merged the strincgs on the same line, they look not so good now tbh

File: dff/script/core/context.py [hunk 8]
```diff
@@ -216,36 +217,36 @@ def last_label(self) -> Optional[NodeLabel2Type]:
...
         return self.responses.get(last_index)
 
-    def set_last_response(self, response: Optional[Any]):
+    def set_last_response(self, response: Optional[Message]):
         """Sets the last `response` of the current :py:class:`~dff.core.engine.core.context.Context`.
         Required for use with various response wrappers.
         """
         last_index = get_last_index(self.responses)
-        self.responses[last_index] = response
+        self.responses[last_index] = Message() if response is None else response
```
[the remark this answers is not in the document]
Comment (RLKRo) [reply dff/script/core/context.py:233]: `Context.responses` values are supposed to be of type `Message`. (If not, we'd have to add an additional check every time we use a response).
I think it's better to make function argument type just `Message`, not `Optional`.

File: examples/pipeline/6_custom_messenger_interface.py [hunk 9]
```diff
@@ -100,17 +100,20 @@ def construct_webpage_by_response(response: str) -> str:
 
 
 def purify_request(ctx: Context):
-    if isinstance(ctx.last_request, Request):
-        logger.info(f"Capturing request from: {ctx.last_request.base_url}")
-        ctx.last_request = ctx.last_request.args.get("request")
-    elif isinstance(ctx.last_request, str):
+    last_request = ctx.last_request
```
[the remark this answers is not in the document]
Comment (RLKRo) [reply examples/pipeline/6_custom_messenger_interface.py:103]: Only for typing. Type hinters and others (linters) assume that value of call `ctx.last_request` may change during execution of `purify_request`.

File: examples/script/core/3_responses.py [hunk 10]
```diff
@@ -43,115 +42,132 @@
```
Comment (RLKRo) [inline examples/script/core/3_responses.py:42]: This description should be changed.

**Comment by kudep** [comment]
All issues of this PR will be solved in others PR

**Review by kudep — APPROVED** [review]

**Pull request merged**